pkg/serial/handshake.go

```diff
@@ -56,19 +54,16 @@ func PurgeIncoming(conn net.Conn) {
 	conn.SetReadDeadline(time.Time{})
 }
 
+func nextPos(syncPos byte) byte { return (syncPos + 1) | 0x80 }
+
 func HandshakeClient(ctx context.Context, conn net.Conn, debug bool) error {
 	if tracing {
 		defer trace.End(trace.Begin(""))
 	}
-	syn := make([]byte, 2)
-	synack := make([]byte, 2)
-	ack := make([]byte, 2)
-
-	buf := make([]byte, 255)
 
-	syn[0] = flagSyn
-	synack[0] = flagAck
-	ack[0] = flagAck
+	buf1byte := make([]byte, 1)
+	rand.Read(buf1byte)
```

Comment:
it easily can happen if reader is closed or times out. I would not panic if it is not critical.